### PR TITLE
HBASE-24467 Backport HBASE-23963: Split TestFromClientSide; it takes …

### PR DESCRIPTION
**Pull request by infraio**
…too long to complete timing out

### CODE REVIEW

File: hbase-common/src/test/java/org/apache/hadoop/hbase/TableNameTestRule.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ */
+package org.apache.hadoop.hbase;
+
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+
+/**
+ * Returns a {@code TableName} based on currently running test method name.
+ */
+public class TableNameTestRule extends TestWatcher {
```
Comment (Apache9) [inline hbase-common/src/test/java/org/apache/hadoop/hbase/TableNameTestRule.java:26]: Good.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 36s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 9 new or modified test files.  |
||| _ branch-2.2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m  4s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   1m 20s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   1m 45s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   4m  2s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 58s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   3m 17s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m  3s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 36s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 20s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 20s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 18s |  hbase-server: The patch generated 12 new + 0 unchanged - 48 fixed = 12 total (was 48)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m  3s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  25m  5s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 2.10.0 or 3.1.2 3.2.1.  |
| +1 :green_heart: |  javadoc  |   0m 56s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m  4s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m  5s |  hbase-common in the patch passed.  |
| -1 :x: |  unit  | 166m 10s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m  8s |  The patch does not generate ASF License warnings.  |
|  |   | 237m  2s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestFromClientSideWithCoprocessor5 |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.10 Server=19.03.10 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1805/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1805 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 7d39baaf6ce5 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1805/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / 37b806484f |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1805/1/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1805/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1805/1/testReport/ |
| Max. process+thread count | 5087 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1805/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by saintstack** [comment]
Rerunning while @infraio  is sleeping.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 49s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 9 new or modified test files.  |
||| _ branch-2.2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   7m  1s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   1m 41s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   2m 33s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   5m 47s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 15s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   4m 36s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   5m 55s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   6m 19s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 24s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 24s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 17s |  hbase-server: The patch generated 12 new + 0 unchanged - 48 fixed = 12 total (was 48)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   3m 59s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  25m 12s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 2.10.0 or 3.1.2 3.2.1.  |
| +1 :green_heart: |  javadoc  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 16s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m  6s |  hbase-common in the patch passed.  |
| -1 :x: |  unit  | 166m 19s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m  8s |  The patch does not generate ASF License warnings.  |
|  |   | 246m 29s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestFromClientSideWithCoprocessor5 |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.10 Server=19.03.10 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1805/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1805 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 4cc953455f44 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1805/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / 604d7bdfb8 |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1805/2/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1805/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1805/2/testReport/ |
| Max. process+thread count | 5071 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1805/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by infraio** [comment]
The failed ut related. Let me check it.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m  3s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 9 new or modified test files.  |
||| _ branch-2.2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m  6s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   1m 19s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   1m 48s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   4m  2s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 58s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   3m 16s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m  1s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 41s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 18s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 18s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 19s |  hbase-server: The patch generated 3 new + 0 unchanged - 48 fixed = 3 total (was 48)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m  3s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  24m 53s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 2.10.0 or 3.1.2 3.2.1.  |
| +1 :green_heart: |  javadoc  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 20s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m  6s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  | 155m 48s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  7s |  The patch does not generate ASF License warnings.  |
|  |   | 228m 25s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.11 Server=19.03.11 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1805/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1805 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 43911d9c543a 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1805/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / 2b1ac7fa4a |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1805/3/artifact/out/diff-checkstyle-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1805/3/testReport/ |
| Max. process+thread count | 4570 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1805/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by infraio** [comment]
Ok. All ut passed. Fix checkstyle problem and retry.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m  3s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 9 new or modified test files.  |
||| _ branch-2.2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 19s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m  7s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   1m 19s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   1m 46s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   4m  4s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 58s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   3m 20s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m  5s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 42s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 21s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 21s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 26s |  The patch passed checkstyle in hbase-common  |
| +1 :green_heart: |  checkstyle  |   1m 16s |  hbase-server: The patch generated 0 new + 0 unchanged - 48 fixed = 0 total (was 48)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m  0s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  25m  8s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 2.10.0 or 3.1.2 3.2.1.  |
| +1 :green_heart: |  javadoc  |   0m 54s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 19s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m  7s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  | 160m  3s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  9s |  The patch does not generate ASF License warnings.  |
|  |   | 233m  1s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.11 Server=19.03.11 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1805/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1805 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 0bf4f4888008 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1805/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / 2b1ac7fa4a |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1805/4/testReport/ |
| Max. process+thread count | 4643 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1805/4/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**